src/components/scroll-monitor.tsx

```diff
@@ -2,48 +2,58 @@ import * as React from 'react'
 import { findDOMNode } from 'react-dom'
 
 interface Props {
+  page: number
   onPage: (number: number) => void
 }
 
+const noop = () => {}
+
 class ScrollMonitor extends React.Component<Props, null> {
-  page = 0
-  threshold = 0
-  teardown: () => void
+  teardown: () => void = noop
```

Comment:
I might opt for just inlining the noop. Then you don't really need the type declaration, right?  Inference would take care of that for you.